app/webpacker/lib/requests/routes.js.erb

```diff
@@ -326,3 +326,25 @@ export const anonymizePersonUrl = `<%= CGI.unescape(Rails.application.routes.url
 
 export const reassignConnectedWcaIdUrl = `<%= CGI.unescape(Rails.application.routes.url_helpers.admin_reassign_wca_id_path) %>`;
 
+export const recordsUrl = (eventId, region, gender, show) => {
+  const url = `<%= CGI.unescape(Rails.application.routes.url_helpers.records_path) %>`
+  const queryParams = new URLSearchParams();
+
+  if (region !== 'world') {
+    queryParams.append('region', region);
+  }
+
+  if (gender !== 'All') {
+    queryParams.append('gender', gender);
+  }
+
+  if (eventId){
+    queryParams.append('event_id', eventId)
+  }
+
+  if (show){
+    queryParams.append('show', show)
+  }
```

Comment:
See my comment on the Rankings PR about redundant-but-not-redundant parameters ;)